web/src/components/FileUpload.jsx

```diff
@@ -48,32 +66,28 @@ export default function FileUpload() {
     setDefaultFileList(fileList);
   };
 
-  const handleCancel = () => {
-    setPreviewVisible(false);
-  };
-
   const handlePreview = async file => {
-    setPreviewVisible(true);
     getBase64(file.originFileObj).then(f => setPreviewImage(f));
+    /**  If you call getBase64(file) you get parameter must be 'blob'. 
+     * If you try to store it with [getBase64(file.originFileObj)] you get the same result on line 62
+     * The commented out lines do not throw exceptions but they do not produce the same key, thus you get an empty result
+     */
+    //TODO: getBase64(file.originFileObj).then(f => setPreviewJSON(annotationResult[md5(f)]));
+    setPreviewJSON(annotationResult[md5(file.name)])
```

Comment:
At least can't we rename the file if we already have a file with the same name as a temprorary solution?

Reply:
Changed the keys for the dictionaries to 'file.uid' from 'file.name'. Program is capable of processing files with the same name.

Reply:
`const handlePreview = async file => { 
    setPreviewImage(previewImages[file.originFileObj.uid])
    setPreviewJSON(annotationResult[file.originFileObj.uid])
  };`